### PR TITLE
Update RFC template

### PR DESCRIPTION
**Pull request by JonasKruckenberg**
This updates the RFC template to streamline writing, give important consideration points more space and deduplicate sections. It also adds more examples and hints for authors on each section.

### CODE REVIEW

File: 0000-template.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+- Feature Name: (fill me in with a unique ident, `my_awesome_feature`)
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- RFC PR: [tauri-apps/rfcs#0000](https://github.com/tauri-apps/rfcs/pull/0000)
+- Discussion: [tauri-apps/rfcs#0000](https://github.com/tauri-apps/rfcs/discussions/0000)
```
Comment (lorenzolewis) [inline 0000-template.md:4]: Should this be changed to "tracking issue" where after a PR is accepted/merged additional conversations/impls happen there?

File: 0000-template.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+# Guide-level explanation
+
+- Explaining the feature largely in terms of examples.
+- If applicable, provide sample error messages, deprecation warnings, or migration guidance.
+- If applicable, describe how this feature can be taught to existing and new Tauri developers.
+  
+# Reference-level explanation
+
+The technical portion of the RFC. Please explain in gory detail:
```
Comment (lorenzolewis) [inline 0000-template.md:22]:
```suggestion
The technical portion of the RFC. Please explain in detail:
```

File: 0000-template.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+- That it is reasonably clear how the feature would be implemented.
+- Any corner cases should be covered by examples.
+- Does this require stale dependencies to be forked? What is the long-term impact on core-developers? Does this introduce additional maintenance load? 
+
+The section should return to the examples given in the previous section, and explain more fully how the detailed proposal makes those examples work.
+
+## Platform Considerations
+
+Tauri works across a wide variety of platforms (Operating Systems, Architectures, Hardware Configurations). Please explain:
+- If applicable, how this feature interacts with different platforms and wether some platforms introduce special behavior.
```
Comment (lorenzolewis) [inline 0000-template.md:34]:
```suggestion
- If applicable, how this feature interacts with different platforms and whether some platforms introduce special behavior.
```

File: 0000-template.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+How will this feature impact the performance of a Tauri app?
+
+## Community Considerations
+
+How does this impact community libraries, tools and other work? Does this invalidate existing tutorials and learning materials? 
+
+# Drawbacks
+
+Why should we *not* do this?
```
Comment (lorenzolewis) [inline 0000-template.md:54]: Maybe rephrase to "What are the limitations of this?" 🤔

File: texts/0004-ipc-router.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
+- Feature Name: `ipc_router`
```
Comment (lorenzolewis) [inline texts/0004-ipc-router.md:1]: I wonder if this file should be removed? At first glance it looks like this is the actual RFC for the IPC router with this file (and wouldn't be merged as-is I'm assuming)

File: 0000-template.md (hunk 1 above)
Comment (lorenzolewis) [reply 0000-template.md:4]: @wusyong suggested making the tracking mechanism more flexible. So instead of only being a tracking issue like I suggested also allowing an implementation PR to track status (for example if a PR is put in prior to an RFC being created).

Comment (JonasKruckenberg) [reply 0000-template.md:4]: I think we can limit it to a tracking issue since any implementation specific PRs and other issues will be linked from that. No need to bloat the RFC template.

Comment (JonasKruckenberg) [reply 0000-template.md]: should be good now?

Comment (lorenzolewis) [reply 0000-template.md]: His point was what if you have impl PR first and then decide you need to actually do an RFC before merging. Then in that case you wouldn't necessarily need a tracking issue. But I think as long as we're flexible with whatever as this is our first pass let's go with as-is so we can get something useable
